source/docs/software/commandbased/commands-v3/mechanisms.rst

```diff
@@ -0,0 +1,434 @@
+     private final PIDController pid = new PIDController(1.0, 0, 0);
+     private final ArmFeedforward feedforward = new ArmFeedforward(0.5, 0.3, 0.1);
+
+     private double targetAngle = 0;
+
+     public Arm() {
+       encoder.setDistancePerPulse(2 * Math.PI / 4096); // radians per tick
+
+       setDefaultCommand(
+         runRepeatedly(() -> holdPosition())
+           .withPriority(Command.LOWEST_PRIORITY)
+           .named("Arm[HOLD]")
+       );
+     }
+
+     // Public sensor reading methods
+     public double getAngle() {
+       return encoder.getDistance();
+     }
+
+     public boolean atGoal() {
+       return Math.abs(getAngle() - targetAngle) < 0.05; // 0.05 radians
+     }
+
+     // Private hardware control methods
+     private void setVoltage(double volts) {
+       motor.setVoltage(volts);
+     }
+
+     private void holdPosition() {
+       double pidOutput = pid.calculate(getAngle(), targetAngle);
+       double ffOutput = feedforward.calculate(targetAngle, 0);
+       setVoltage(pidOutput + ffOutput);
+     }
+
+     // Command: Move to specific angle
+     public Command moveTo(double angle) {
+       return run(coroutine -> {
+         targetAngle = angle;
+         while (!atGoal()) {
+           holdPosition(); // Use PID + feedforward
+           coroutine.yield();
+         }
+       }).named("Arm Move to " + angle);
+     }
+
+     // Command: Manual control
+     public Command manualControl(DoubleSupplier speedSupplier) {
+       return runRepeatedly(() -> {
+         double speed = speedSupplier.getAsDouble();
+         targetAngle = getAngle(); // Track current position
+         setVoltage(speed * 12.0); // Direct voltage control
+       }).named("Arm Manual");
+     }
+
+     // Command: Home the arm
+     public Command home() {
+       return run(coroutine -> {
+         // Move down slowly until limit switch
+         while (!limitSwitch.get()) {
+           setVoltage(-2.0);
+           coroutine.yield();
+         }
+         encoder.reset();
+         targetAngle = 0;
+         setVoltage(0);
+       }).named("Home Arm");
+     }
+
+     public Arm() {
+       // Add telemetry via periodic hook
+       Scheduler.getDefault().addPeriodicHook(() -> {
+         Telemetry.log("Arm/Angle", getAngle());
+         Telemetry.log("Arm/Target", targetAngle);
+         Telemetry.log("Arm/At Goal", atGoal());
+       });
+
+       // Set default command
+       setDefaultCommand(
+         runRepeatedly(() -> setVoltage(0))
+           .withPriority(Command.LOWEST_PRIORITY)
+           .named("Arm[IDLE]")
+       );
+     }
+   }
+
+## Best Practices
+
+1. **Keep hardware private**: Expose methods, not hardware objects. This lets you change hardware without changing commands.
+
+2. **Use descriptive method names**: ``tank(left, right)`` is clearer than ``set(l, r)``
+
+3. **Publish telemetry with periodic hooks or Epilogue**: Don't clutter command code with dashboard calls
```

Comment:
This also has the advantage that the telemetry is always available, not just when a command is running